### ISSUE TITLE
Appdomain.TypeResolve event being raised for "Mono.Runtime" while Logging everytime

### ISSUE THREAD

**Issue by ChaitanyaKN** [NLog/NLog#2416]
Type: Bug

**NLog version**: (e.g. 4.4.12)

**Platform**: .Net 3.5 

**Current NLog config** (xml or C#, if relevant)
 <nlog >
   <extensions>
     <add assembly ="MyAssembly"/>
   </extensions>
  
 <targets>
  <target name="TargFile" xsi:type="CFile" fileName="c:/logs/Test.txt" />  
 </targets>

   <rules>
     <logger name="*" minlevel="Info" writeTo="TargFile"/>
   </rules>    
 </nlog>
In case of a BUG:

- What is the current result?
LOGGING IS WORKING AS EXPECTED but:
I am getting a ResolveType Exception(I Subscribed to AppDomain.TypeResolve) whenever I try ".log"(or.info,.error etc). It seems the Exception occurs whenever Type.GetType(Mono.Runtime) is called.  

- What is the expected result?
I expect  ResolveType event to not be called.

- Did you checked the [Internal log](https://github.com/NLog/NLog/wiki/Internal-Logging)?
No new messages appeared in Internal log.

- Please post full exception details (message, stacktrace, inner exceptions)
I believe the class "Nlog/Internal/PlatformDetector.cs" is responsible. The ResolveType Exception only has one property : ex.Name,this property says the Type Failing to resolve is "Mono.Runtime". 

 
- Are there any work arounds? By default no one subscribes to Appdomain.TypeResolve, and thus they will not notice the event, that might not exactly be a workaround as i need to listen to Typeresolve from other parts of my code.

- Is there a version in which it did worked? 
Logging Works, but this event is being raised everytime.

Is there any way to prevent PlatformDetector to check for Mono.Runtime type everytime i try logging?My application needs to check for any typeresolve events as well as a lot of others and log them too, but my console and log file is filled with Typeresolve for "Mono.runtime" after every line.

**Comment by 304NotModified** [comment]
what's your version of NLog? The exception should be in the internal log - please post the full details (stack trace etc)

**Comment by ChaitanyaKN** [comment]
**logging is working fine**, and there is nothing being logged in internal log. I am getting the ResolveEventArgs exception only because I created a method that listens to Typeresolve.

_ad.TypeResolve += new ResolveEventHandler(_ad_TypeResolve);(TypeResolve is an event https://msdn.microsoft.com/en-us/library/system.appdomain.typeresolve(v=vs.90).aspx 
_ad is my AppDomain.
)
_ad_Typeresolve is my own method that handles the event.
Assembly _ad_TypeResolve(object sender, ResolveEventArgs args) {
            log.Warn("{0} Failed to TypeResolve for Assembly:{1}", _ad.FriendlyName, args.Name);
            return null;
	}
The message that is triggered in my console is "MyAppDomain" Failed to TypeResolve for Assembly: Mono.Runtime. If i comment my own method out, then nlog works fine.

The only place where I've seen "Mono.runtime" being referred to is in PlatformDetector.cs. Its not in my code, but is in Nlog/Internal/PlatformDetector.cs.
In this line:
public static bool IsMono => Type.GetType("Mono.Runtime") != null;

 The same Typeresolve event was raised when I inserted the line 
Type.GetType("Mono.Runtime") into my own code without Nlog. 
 
I am catching this exception only because I subscribed to such events, or it would be ignored.  I might not be right in saying this but I believe Nlog is not causing an error its simply calling Type.Gettype() which raises an event, and thus these should not be logged in internal log. I just need a workaround that doesnt call Type.GetType Every time.
